packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -25,17 +27,21 @@ function NavigatorProvider(
 	props: WordPressComponentProps< NavigatorProviderProps, 'div' >,
 	forwardedRef: Ref< any >
 ) {
-	const { initialPath, children, ...otherProps } = useContextSystem(
-		props,
-		'NavigatorProvider'
-	);
+	const {
+		initialPath,
+		children,
+		className,
+		...otherProps
+	} = useContextSystem( props, 'NavigatorProvider' );
 
 	const [ path, setPath ] = useState< NavigatorPath >( {
 		path: initialPath,
 	} );
 
+	const classes = useCx()( css( { overflowX: 'hidden' } ), className );
```

Comment:
Let's memoise classname computation

```suggestion
	const cx = useCx();
	const classes = useMemo(
		() => cx( css( { overflowX: 'hidden' } ), className ),
		[ className ]
	);
```